core/src/main/java/org/apache/iceberg/util/PartitionSet.java

```diff
@@ -200,7 +200,8 @@ public String toString() {
           StringBuilder partitionStringBuilder = new StringBuilder();
           partitionStringBuilder.append(structType.fields().get(i).name());
           partitionStringBuilder.append("=");
-          partitionStringBuilder.append(s.get(i, Object.class).toString());
+          Object value = s.get(i, Object.class);
+          partitionStringBuilder.append(value == null ? "null" : value.toString());
```

Comment:
please use `String.valueOf(s.get(i, Object.class))` instead

Reply:
Yeah, that's nicer. Done.